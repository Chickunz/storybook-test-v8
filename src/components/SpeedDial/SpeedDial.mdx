import { Meta, Canvas, ArgTypes } from "@storybook/blocks";
import * as SpeedDialStories from "./SpeedDial.stories";

<Meta of={SpeedDialStories} />

<style>
  {`
    .subheading {
      --mediumdark: '#999999';
      font-weight: 700;
      font-size: 13px;
      color: #999;
      letter-spacing: 6px;
      line-height: 24px;
      text-transform: uppercase;
      margin-bottom: 12px;
      margin-top: 40px;
    }
  `}
</style>

# SpeedDial Component

Making My Components Here

<div className="subheading">SpeedDial</div>

<div>This is using the up SpeedDial!</div>

<Canvas of={SpeedDialStories.Up} />

<ArgTypes of={SpeedDialStories} />

<div>This is the speed dial but right!!</div>

<Canvas of={SpeedDialStories.Right} />
