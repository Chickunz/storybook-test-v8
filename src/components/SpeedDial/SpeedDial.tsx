import React from "react";
import { SpeedDial as MUISpeedDial } from "@mui/material";

export interface SpeedDialProps
  extends React.HTMLAttributes<HTMLButtonElement> {
  /**
   * Which way is the icon facing
   */
  direction?: "down" | "left" | "right" | "up";
  /**
   * Arialabel for the component
   */
  ariaLabel: string;
}

export const SpeedDial = React.forwardRef(
  ({ direction = "up", ariaLabel }: SpeedDialProps, ref) => {
    return (
      <MUISpeedDial direction={direction} ariaLabel={ariaLabel} ref={ref} />
    );
  }
);

SpeedDial.displayName = "SpeedDial";

export default SpeedDial;
