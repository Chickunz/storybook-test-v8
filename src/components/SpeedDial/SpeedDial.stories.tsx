import SpeedDial from "./SpeedDial";

export default {
  title: "Components/SpeedDial",
  component: SpeedDial
};

// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Up = {
  args: {
    ariaLabel: "Hai"
  }
};

export const Right = {
  args: {
    direction: "right",
    ariaLabel: "Hai but right"
  }
};
