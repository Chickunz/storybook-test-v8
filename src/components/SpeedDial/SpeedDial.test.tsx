import React from "react";
import SpeedDial from "./SpeedDial";
import { describe, it, expect, beforeEach } from "vitest";
import { render } from "@testing-library/react";

describe("SpeedDial Test Cases", () => {
  let getByRole: (arg0: string) => HTMLElement,
    asFragment: () => DocumentFragment;

  beforeEach(() => {
    ({ getByRole, asFragment } = render(<SpeedDial ariaLabel={"Testing"} />));
  });

  it("Should Match SpeedDial Snapshot", () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
