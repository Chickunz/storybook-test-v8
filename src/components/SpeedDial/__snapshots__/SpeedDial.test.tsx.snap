// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SpeedDial Test Cases > Should Match SpeedDial Snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiSpeedDial-root MuiSpeedDial-directionUp css-h0fz3e-MuiSpeedDial-root"
    role="presentation"
  >
    <button
      aria-controls="Testing-actions"
      aria-expanded="false"
      aria-haspopup="true"
      aria-label="Testing"
      class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-primary MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-primary MuiSpeedDial-fab css-7dv1rb-MuiButtonBase-root-MuiFab-root-MuiSpeedDial-fab"
      style="transform: none; webkit-transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <div
      aria-orientation="vertical"
      class="MuiSpeedDial-actions MuiSpeedDial-actionsClosed css-1s28u9o-MuiSpeedDial-actions"
      id="Testing-actions"
      role="menu"
    />
  </div>
</DocumentFragment>
`;
